modules/project/alert-metrics.tf

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  _logging_metrics_alerts_factory_data_raw = merge([
+    for f in try(fileset(var.factories_config.logging_metrics_alerts, "*.yaml"), []) :
+    yamldecode(file("${var.factories_config.logging_metrics_alerts}/${f}"))
+  ]...)
+  _logging_metrics_alerts_factory_data = {
+    for k, v in local._logging_metrics_alerts_factory_data_raw :
+    k => merge({
+      name                  = k
+      description           = null
+      filter                = null
+      display_name          = null
+      metric_descriptor     = {}
+      condition_threshold   = {}
+      trigger_count         = null
+      aggregations          = {}
+      combiner              = null
+      notification_channels = []
+    }, v)
+  }
+  logging_metrics_alerts = merge(local._logging_metrics_alerts_factory_data, var.logging_metrics_alerts)
+}
+
+resource "google_monitoring_notification_channel" "default" {
+  project      = local.project.project_id
+  display_name = "Default Email Notification"
+  type         = "email"
+  labels = {
+    email_address = try(var.default_alerts_email)
+  }
+}
+
+resource "google_logging_metric" "default" {
```

Comment:
This only allows the creation of the simplest metrics. The resource (`google_logging_metric`) provides much more flexibility which should be expose by the factory.